### DISCUSSION TITLE
List of supported browsers and versions?

### DISCUSSION THREAD

**Discussion by klickreflex** [alpinejs/alpine#2901] [1. Help]
Hey everybody,

I've recently received numerous clients' complaints about broken functionality on their not-so-recent devices and browsers. 

I totally understand and like that Alpine doesn't want to keep legacy baggage forever, but I'd really love to see some sort of overview about supported browser version and, if applicable, even links to existing Polyfills (such as for [queueMicrotasks](https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask#when_queuemicrotask_isnt_available)) that can help to keep our sites accessible for people who don't buy new devices every other year.

If such a list already does exist, I couldn't figure out where to find it 🙃

**Comment by philwolstenholme** [comment]
This isn't quite what you are asking for, but in the past I have put this in the `<head>`. The `nomodule` stops [newer browsers](https://caniuse.com/?search=es6%20modules) from downloading/executing the file, but for older browsers it sends a bunch of polyfills depending on how old the browser is. Alpine won't need all the features listed in the URL, but I added them all in to cover other code I had written.

You can find out more about the Polyfill.io service here: https://polyfill.io/v3/

```
<script src="//polyfill.io/v3/polyfill.min.js?flags=gated&features=default,es5,es6,es7,matchMedia,IntersectionObserver,ResizeObserver,NodeList.prototype.forEach,HTMLTemplateElement,Element.prototype.closest,requestAnimationFrame,CustomEvent,URLSearchParams,queueMicrotask" nomodule></script>
```

**Reply by klickreflex** [reply]
Thanks @philwolstenholme,

That's really helpful in some cases  👍 

Unfortunately, it doesn't work always: e.g. `queueMicrotask` is supported as of FF 69, but FF supports es modules starting with v60, so 60-68 won't load the polyfills using nomodules.

**Reply by philwolstenholme** [reply]
Ah, well in that case you can just drop the `nomodule` and accept that there will be a slight performance hit from making a HTTP connection to `https://polyfill.io`. For up-to-date browsers, the response will come back empty (try visiting https://polyfill.io/v3/polyfill.min.js?flags=gated&features=default,es5,es6,es7,matchMedia,IntersectionObserver,ResizeObserver,NodeList.prototype.forEach,HTMLTemplateElement,Element.prototype.closest,requestAnimationFrame,CustomEvent,URLSearchParams,queueMicrotask in the latest Chrome, for example) so the performance impact will be minimal, just the cost of a DNS lookup, HTTPS negotiation and that sort of thing.
